docs/design/aws_under_the_hood.md

```diff
@@ -0,0 +1,271 @@
+a distribution file, and then it is responsible for attaching and detaching EBS
+volumes to itself.
+
+The minion policy is relatively minimal.  The master policy is probably overly
+permissive.  The security concious may want to lock-down the IAM policies
+further (#11936)
+
+We should make it easier to extend IAM permissions and also ensure that they
+are correctly configured (#???)
+
+### Tagging
+
+All AWS resources are tagged with a tag named "KuberentesCluster".  This tag is
+used to identify a particular 'instance' of Kubernetes, even if two clusters
+are deployed into the same VPC.  (The script doesn't do this by default, but it
+can be done.)
+
+Within the AWS cloud provider logic, we filter requests to the AWS APIs to
+match resources with our cluster tag.  So we only see our own AWS objects.
+
+If you choose not to use kube-up, you must tag everything with a
+KubernetesCluster tag with a unique per-cluster value.
+
+
+# AWS Objects
+
+The kube-up script does a number of things in AWS:
+
+* Creates an S3 bucket (`AWS_S3_BUCKET`) and copy the kubernetes distribution
+  and the salt scripts into it.  They are made world-readable and the HTTP URLs
+are passed to instances; this is how kubernetes code gets onto the machines.
+* Creates two IAM profiles based on templates in `cluster/aws/templates/iam`.
+  `kubernetes-master` is used by the master node; `kubernetes-minion` is used
+by minion nodes.
+* Creates an AWS SSH key named `kubernetes-<fingerprint>`.  Fingerprint here is
+  the OpenSSH key fingerprint, so that multiple users can run the script with
+different keys and their keys will not collide (with near-certainty) It will
+use an existing key if one is found at `AWS_SSH_KEY`, otherwise it will create
+one there.  (With the default ubuntu images, if you have to SSH in: the user is
+`ubuntu` and that user can `sudo`)
+* Creates a VPC for use with the cluster (with a CIDR of 172.20.0.0/16)., and
+  enables the `dns-support` and `dns-hostnames` options.
+* Creates an internet gateway for the VPC.
+* Creates a route table for the VPC, with the internet gateway as the default
+  route
+* Creates a subnet (with a CIDR of 172.20.0.0/24) in the AZ `KUBE_AWS_ZONE`
+  (defaults to us-west-2a).  Currently kubernetes runs in a single AZ; there
+are two philosophies on how to achieve HA: cluster-per-AZ and
+cross-AZ-clusters.  cluster-per-AZ says you should have an independent cluster
+for each AZ, they are entirely separate.  cross-AZ-clusters allows a single
+cluster to span multiple AZs.  The debate is open here: cluster-per-AZ is more
+robust but cross-AZ-clusters are more convenient.  For now though, each AWS
+kuberentes cluster lives in one AZ.
+* Associates the subnet to the route table
+* Creates security groups for the master node (`kubernetes-master-<clusterid>`)
+  and the minion nodes (`kubernetes-minion-<clusterid>`)
+* Configures security groups so that masters & minions can intercommunicate,
```

Comment:
Edit (lines 200-202):

Configures security groups so that masters and nodes can communicate. This includes intercommunication between masters and nodes, opening SSH publicly for both masters and nodes, and opening port 443 on the master for the HTTPS API endpoints.